### PR TITLE
Drupal Rogue client

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?

Adds functions to the new `dosomething_rogue` module that connects to the rogue api and sets up the wrapper for the guzzle client that makes the actual request.
#### How should this be reviewed?

You should be able to call `ddosomething_rogue_send_reportback_to_rogue()` and receive a response from rogue. 
#### Any background context you want to provide?

Unblocks work on sending reportbacks and updates to reportbacks.
#### Relevant tickets

Ref #6960 #6972

### CODE REVIEW

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module [hunk 1]
```diff
@@ -29,3 +29,58 @@ function dosomething_rogue_menu() {
...
+ * @param array $values
+ *   Values to send to Rogue.
+ * @todo - Do we need to pass in user here? Do we need to pass in the item separately?
+ */
+function dosomething_rogue_send_reportback_to_rogue($values, $user = NULL) {
+  if (!isset($user)) {
+    global $user;
+  }
+
+  $campaign = node_load($values['nid']);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: do you need to load the node here? looks like you just use the `nid` below

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module [hunk 2]
```diff
@@ -29,3 +29,58 @@ function dosomething_rogue_menu() {
...
+          'quantity' => $values['quantity'],
+          'why_participated' => $values['why_participated'],
+          'file' => $values['file'],
+          'caption' => $values['caption'],
+          // @TODO - review status stuff can be passed here also
+        ]),
+  ];
+
+  $response = drupal_http_request($client['base_url'] . '/reportbacks', $options);
+
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module:83]: maybe add a todo here to add some response/error handling

**Review by angaither — CHANGES_REQUESTED** [review]

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module (hunk 1 above)
Comment (sbsmith86) [reply lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: yea my bad, that was a relic of some previous version of this code.

**Review by angaither — APPROVED** [review]

**Comment by angaither** [comment]
🚢

**Pull request merged**